src/librustdoc/html/format.rs

```diff
@@ -456,15 +456,26 @@ impl clean::GenericArgs {
     }
 }
 
-crate fn href(did: DefId, cx: &Context<'_>) -> Option<(String, ItemType, Vec<String>)> {
+// Possible errors when computing href link source for a `DefId`
+crate enum HrefError {
```

Comment:
This would help quite a lot with https://github.com/rust-lang/rust/pull/82014 I think :) good idea.